packages/common/src/cis2/util.ts

```diff
@@ -431,6 +450,70 @@ function serializeCIS2OperatorOfQuery(query: CIS2.OperatorOfQuery): Buffer {
     return Buffer.concat([owner, address]);
 }
 
+/**
+ * Parses a token address from a Base58-string. Will throw if the Base58
+ * encoding is not a valid token address.
+ *
+ * @param str A Base58 encoded token address
+ * @returns A parsed token address
+ */
+export function tokenAddressFromBase58(str: Base58String): CIS2.TokenAddress {
+    const bytes = new Buffer(bs58check.decode(str));
+
+    const firstByte = bytes[0];
+    const [index, i] = uleb128DecodeWithIndex(bytes, 1);
+    const [subindex, j] = uleb128DecodeWithIndex(bytes, i);
```

Comment:
I would not expect a contract address to be leb128 encoded? Where does this come from?

Normally, a contract address is encoded as 8 bytes for index and subindex respectively, meaning 16 bytes for the entire contract address (not including the token ID of course). Check out the `serializeContractAddress` to see how it is serialized, and deserialize accordingly 🙂

Reply:
It comes from the spec, and it must adhere to it, right?

See:
https://proposals.concordium.software/CIS/cis-2.html#token-address

Reply:
Yes, sorry 🙈